qa/tasks/cephfs/test_snap_schedules.py

```diff
@@ -577,6 +579,400 @@ def test_snap_schedule_all_periods(self):
         self.mount_a.run_shell(['rmdir', test_dir])
 
 
+class TestSnapSchedulesSubvolAndGroupArguments(TestSnapSchedulesHelper):
+    def setUp(self):
+        super(TestSnapSchedulesSubvolAndGroupArguments, self).setUp()
+        self.CREATE_VERSION = (random.randrange(10) % 2) + 1
+        self._disable_subvolume_upgrade()
+
+    def _create_v1_subvolume(self, subvol_name, subvol_group=None, has_snapshot=False, subvol_type='subvolume', state='complete'):
+        group = subvol_group if subvol_group is not None else '_nogroup'
+        basepath = os.path.join("volumes", group, subvol_name)
+        uuid_str = str(uuid.uuid4())
+        createpath = os.path.join(basepath, uuid_str)
+        self.mount_a.run_shell(['sudo', 'mkdir', '-p', createpath], omit_sudo=False)
+
+        # create a v1 snapshot, to prevent auto upgrades
+        if has_snapshot:
+            snappath = os.path.join(createpath, ".snap", "fake")
+            self.mount_a.run_shell(['sudo', 'mkdir', '-p', snappath], omit_sudo=False)
+
+        # add required xattrs to subvolume
+        default_pool = self.mount_a.getfattr(".", "ceph.dir.layout.pool")
+        self.mount_a.setfattr(createpath, 'ceph.dir.layout.pool', default_pool, sudo=True)
+
+        # create a v1 .meta file
+        cp = "/" + createpath
+        meta_contents = f"[GLOBAL]\nversion = 1\ntype = {subvol_type}\npath = {cp}\nstate = {state}\n"
+        if state == 'pending':
+            # add a fake clone source
+            meta_contents = meta_contents + '[source]\nvolume = fake\nsubvolume = fake\nsnapshot = fake\n'
+        meta_filepath1 = os.path.join(self.mount_a.mountpoint, basepath, ".meta")
+        self.mount_a.client_remote.write_file(meta_filepath1, meta_contents, sudo=True)
+        return createpath
+
+    def _create_subvolume(self, version, subvol_name, subvol_group=None):
+        if version == 1:
+            self._create_v1_subvolume(subvol_name, subvol_group)
+        elif version == 2:
+            self._fs_cmd('subvolume', 'create', 'cephfs', subvol_name, '--group_name', subvol_group)
+        else:
+            self.assertTrue('NoSuchSubvolumeVersion' == None)
+
+    def _disable_subvolume_upgrade(self):
+        # this will also disable subvolume upgrades
+        self.config_set('mgr', 'mgr/volumes/allow_subvolume_upgrade', False)
+
+    def test_snap_schedule_subvol_and_group_arguments_01(self):
```

Comment:
This should additionally verify if the schedule snapshots have been taken in the respective subvolume snapshot directory (for v1 vs v2 subvolumes).

Reply:
this needs some work to confirm snapshot creation

Reply:
Agree - but IMO it would greatly benefit the feature testing.